samples/apps/logging/logging.cpp

```diff
@@ -318,10 +318,8 @@ namespace loggingapp
         static constexpr auto CCF_TX_ID = "x-ms-ccf-transaction-id";
         ctx.rpc_ctx->set_response_header(CCF_TX_ID, tx_id.to_str());
 
-        const nlohmann::json body_j =
-          nlohmann::json::parse(ctx.rpc_ctx->get_response_body());
-
-        auto interm = body_j.get<LoggingPut::Intermediate>();
+        auto interm =
+          *static_cast<LoggingPut::Intermediate*>(ctx.rpc_ctx->get_user_data());
```

Comment:
@eddyashton Is there a nicer way to put this?

Reply:
There's nothing much nicer - we've lost any type safety by casting to `void*`, so we need to `static_cast` _somewhere_. We could hide this if we were doing it a lot, but I think it's fine to be this obvious in a sample. We could also build a polymorphic type hierarchy for the values you're allowed to store as `user_data`, inheriting from a common type so we can `dynamic_cast` back and guarantee we've got the expected type, but I don't think that's worth it.

However, I think the return type should be `Intermediate*` rather than `Intermediate` (ie - remove the `*` before the `static_cast`, replace `interm.` with `interm->`). That way we can return a slightly clearer error if it ends up being null (when some path in the handler _doesn't_ call `set_user_data()`), which is a good thing to include in the sample implementation.